internal/generator/vector/sources.go

```diff
@@ -118,6 +120,81 @@ func ExcludeContainerPaths(namespace string) string {
 	))
 }
 
+func HttpSources(spec *logging.ClusterLogForwarderSpec, op generator.Options) []generator.Element {
+	var minTlsVersion, cipherSuites string
+	if _, ok := op[generator.ClusterTLSProfileSpec]; ok {
+		tlsProfileSpec := op[generator.ClusterTLSProfileSpec].(configv1.TLSProfileSpec)
+		minTlsVersion = tls.MinTLSVersion(tlsProfileSpec)
+		cipherSuites = strings.Join(tls.TLSCiphers(tlsProfileSpec), `,`)
+	}
+
+	el := []generator.Element{}
+	for _, input := range spec.Inputs {
+		if input.Source != nil {
+			if input.Source.HTTP != nil {
+				el = append(el, HttpSource{
+					ID:            input.Source.HTTP.Name,
+					Port:          input.Source.HTTP.Port.Port,
+					Format:        input.Source.HTTP.Format,
+					LogType:       input.Source.HTTP.LogType,
+					TlsMinVersion: minTlsVersion,
+					CipherSuites:  cipherSuites,
+				})
+			}
+		}
+	}
+	return el
+}
+
+type HttpSource struct {
+	ID            string
+	Port          int32
+	Format        string
+	LogType       string
+	TlsMinVersion string
+	CipherSuites  string
+}
+
+func (HttpSource) Name() string {
+	return "httpSource"
+}
+
+func (i HttpSource) Template() string {
+	return `
+{{define "` + i.Name() + `" -}}
+[sources.{{.ID}}]
+type = "http_server"
+address = "0.0.0.0:{{.Port}}"
+decoding.codec = "json"
+
+[sources.{{.ID}}.tls]
+enabled = false
+key_file = "/etc/collector/{{.ID}}/tls.key"
```

Comment:
We need to ensure this is still valid in context of multi CLF.  As long as we are setting up the volumes correctly for the DS this will be fine

Reply:
^^^ https://github.com/openshift/cluster-logging-operator/pull/2155#discussion_r1313485608